### ISSUE TITLE
usdbkey  wm_devicechange

### ISSUE THREAD

**Issue by ycper** [sandboxie-plus/Sandboxie#1975]
### What happened?

Win7 system. No USBKEY insertion message is received in the sandbox, such as

DBT_ Devicearrival and DBT_ DEVICEREMOVECOMPLETE 。

### Download link

Not relevant

### To Reproduce

1

### Expected behavior

I hope the application can receive the unplugging event of USBKEY

### What is your Windows edition and version?

windows 7 

### In which Windows account you have this problem?

Not relevant to my request.

### Please mention any installed security software

No security software installed

### What version of Sandboxie are you running?

1.0.10.0 64

### Is it a regression?

1.0.10.0 64

### List of affected browsers

No browser used

### In which sandbox type you have this problem?

In a Standard isolation sandbox (yellow sandbox icon).

### Where is the program located?

The program is installed both inside and outside the sandbox.

### Can you reproduce this problem on an empty sandbox?

Not relevant to my request.

### Did you previously enable some security policy settings outside Sandboxie?

default Windows settings.

### Sandboxie.ini configuration

```shell
#
# Sandboxie-Plus configuration file
#

[GlobalSettings]
FileRootPath=\??\%SystemDrive%\Sandbox\%USER%\%SANDBOX%
SeparateUserFolders=y
KeyRootPath=\REGISTRY\USER\Sandbox_%USER%_%SANDBOX%
IpcRootPath=\Sandbox\%USER%\%SANDBOX%\Session_%SESSION%
NetworkEnableWFP=n
EnableObjectFiltering=n
EnableWin32kHooks=y
EditAdminOnly=n
ForceDisableAdminOnly=n
ForgetPassword=n
StartRunAlertDenied=y
NotifyStartRunAccessDenied=y
AlertProcess=test.exe
AlertFolder=C:\backup

[DefaultBox]
ConfigLevel=9
BlockNetworkFiles=y
Template=OpenSmartCard
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=Chrome_Phishing_DirectAccess
Template=Firefox_Phishing_DirectAccess
Template=AutoRecoverIgnore
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Desktop%
BorderColor=#00ffff,ttl,6
Enabled=y
ForceProcess=notepad.exe
BoxNameTitle=n
OpenClipboard=n
CopyLimitKb=81920
OpenDevCMApi=y
OpenSamEndpoint=y
OpenLsaEndpoint=y
PipeTrace=*
IpcTrace=*
ClsidTrace=*
ClosedFilePath=<BlockNetDevices>,InternetAccessDevices
#OriginalToken=y

[UserSettings_4BC00582]
SbieCtrl_UserName=administrator
SbieCtrl_BoxExpandedView=DefaultBox
SbieCtrl_NextUpdateCheck=-1
SbieCtrl_WindowCoords=816,225,1104,408
SbieCtrl_ActiveView=40022
SbieCtrl_UpdateCheckNotify=n
SbieCtrl_ExplorerWarn=n
SbieCtrl_ProcessViewColumnWidths=250,70,300
SbieCtrl_AutoStartAgent=SandMan.exe
SbieCtrl_EnableAutoStart=y

[New_Box]
Enabled=y
AutoRecover=y
BlockNetworkFiles=y
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Desktop%
BorderColor=#00ffff,ttl,6
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=Chrome_Phishing_DirectAccess
Template=Firefox_Phishing_DirectAccess
Template=AutoRecoverIgnore
ConfigLevel=9
UsePrivacyMode=n
BoxNameTitle=n
OpenClipboard=n
CopyLimitKb=81920
ForceProcess=test1.ext
ForceProcess=test.exe
```

**Comment by ycper** [comment]
[CreateWindow.txt](https://github.com/sandboxie-plus/Sandboxie/files/8938921/CreateWindow.txt)
------------------
The above is the application source code

**Comment by DavidXanatos** [comment]
strange for me it seams to work tested on win 7 as well

WndProc Entry, message=0x219, HWND = d066a
1:   DeviceChange   wParam=0x7? lParam=0
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
2:   DeviceChange   wParam=0x7? lParam=0
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
3:   DeviceChange   wParam=0x7? lParam=0
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
4:   DeviceChange   wParam=0x7? lParam=0
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
5:   DeviceChange   wParam=0x8000? lParam=0x237fe10
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
6:   DeviceChange   wParam=0x7? lParam=0
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
7:   DeviceChange   wParam=0x7? lParam=0
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
8:   DeviceChange   wParam=0x7? lParam=0
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
9:   DeviceChange   wParam=0x7? lParam=0
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
10:   DeviceChange   wParam=0x7? lParam=0
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
11:   DeviceChange   wParam=0x8004? lParam=0x237fe10
WndProc tid = 4476
WndProc Entry, message=0x219, HWND = d066a
12:   DeviceChange   wParam=0x7? lParam=0

**Comment by liuxianyue** [comment]
I have the same problem,
in sbiedll, registerdevicenotificationW is hooked, GUI_ RegisterdevicenotificationW is not implemented really , only Calls SetLastError(0), and returned 0x12345678 .

Now Modify GUI_Registerdevicenotificationaw，add __sys_RegisterdevicenotificationW , the sandboxed program a still cannot receive any inserting and pulling notifications of the Device..

**Comment by liuxianyue** [comment]
the same  #2129,  I'm looking forward to the perfect answer

**Comment by liuxianyue** [comment]
I haven't paid attention to this aspect for a long time













仙乐飘飘
***@***.***







                       
Original Email
                       
                     

Sender:"DavidXanatos"< ***@***.*** &gt;;

Sent Time:2023/5/26 13:39

To:"sandboxie-plus/Sandboxie"< ***@***.*** &gt;;

Cc recipient:"liuxianyue"< ***@***.*** &gt;;"Comment"< ***@***.*** &gt;;

Subject:Re: [sandboxie-plus/Sandboxie] usdbkey  wm_devicechange (Issue #1975)




 
Closed #1975 as completed.

**Comment by DavidXanatos** [comment]
we have a long back log of known issues so some things sometimes take time, sorry about that....
it would be great if you could test it and confirm the fix.
